Sim issue

---------- Bug report ----------

Please include a description about what actions you were performing when the exception occurred:
(You can edit text directly in this window)

1.Just running the sim, I chaged the mass of the parachutes but the apogee stopped showing. 
2.
3.

If possible, please send us the .ork file that caused the bug.

Include your email address (optional; it helps if we can contact you in case we need additional information):




(Do not modify anything below this line.)
---------- Exception stack trace ----------
info.openrocket.swing.gui.main.SwingExceptionHandler$InternalException: An exception occurred during the simulation at info.openrocket.swing.gui.main.SwingExceptionHandler.handleErrorCondition(SwingExceptionHandler.java:121) at info.openrocket.swing.gui.simulation.SimulationRunDialog$InteractiveSimulationWorker.simulationInterrupted(SimulationRunDialog.java:446) at info.openrocket.swing.gui.simulation.SimulationWorker.done(SimulationWorker.java:109) at java.desktop/javax.swing.SwingWorker$5.run(SwingWorker.java:750) at java.desktop/javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.run(SwingWorker.java:848) at java.desktop/sun.swing.AccumulativeRunnable.run(AccumulativeRunnable.java:112) at java.desktop/javax.swing.SwingWorker$DoSubmitAccumulativeRunnable.actionPerformed(SwingWorker.java:858) at java.desktop/javax.swing.Timer.fireActionPerformed(Timer.java:311) at java.desktop/javax.swing.Timer$DoPostEvent.run(Timer.java:243) at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318) at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773) at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720) at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714) at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742) at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:117) at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:191) at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:236) at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:234) at java.base/java.security.AccessController.doPrivileged(AccessController.java:318) at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:234) at java.desktop/java.awt.Dialog.show(Dialog.java:1080) at java.desktop/java.awt.Component.show(Component.java:1728) at java.desktop/java.awt.Component.setVisible(Component.java:1675) at java.desktop/java.awt.Window.setVisible(Window.java:1036) at java.desktop/java.awt.Dialog.setVisible(Dialog.java:1016) at info.openrocket.swing.gui.main.SimulationPanel.runSimulation(SimulationPanel.java:500) at info.openrocket.swing.gui.main.SimulationPanel$RunSimulationAction.actionPerformed(SimulationPanel.java:1084) at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972) at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313) at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405) at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262) at java.desktop/javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:279) at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297) at java.desktop/java.awt.Component.processMouseEvent(Component.java:6626) at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3389) at java.desktop/java.awt.Component.processEvent(Component.java:6391) at java.desktop/java.awt.Container.processEvent(Container.java:2266) at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5001) at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324) at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833) at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948) at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575) at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516) at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2310) at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780) at java.desktop/java.awt.Component.dispatchEvent(Component.java:4833) at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:775) at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:720) at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:714) at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97) at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:747) at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745) at java.base/java.security.AccessController.doPrivileged(AccessController.java:399) at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:744) at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113) at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109) at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90) Caused by: info.openrocket.core.util.BugException: BUG: Simulation resulted in not-a-number (NaN) value for structureMass.getCenterOfMass(), please report a bug, c=(NaN,NaN,NaN,w=6.50156) at info.openrocket.core.simulation.AbstractSimulationStepper.checkNaN(AbstractSimulationStepper.java:266) at info.openrocket.core.simulation.AbstractSimulationStepper.calculateStructureMass(AbstractSimulationStepper.java:217) at info.openrocket.core.simulation.RK4SimulationStepper.computeAcceleration(RK4SimulationStepper.java:337) at info.openrocket.core.simulation.RK4SimulationStepper.calculateAcceleration(RK4SimulationStepper.java:284) at info.openrocket.core.simulation.RK4SimulationStepper.computeParameters(RK4SimulationStepper.java:261) at info.openrocket.core.simulation.RK4SimulationStepper.step(RK4SimulationStepper.java:113) at info.openrocket.core.simulation.BasicEventSimulationEngine.simulateLoop(BasicEventSimulationEngine.java:198) at info.openrocket.core.simulation.BasicEventSimulationEngine.simulate(BasicEventSimulationEngine.java:121) at info.openrocket.core.document.Simulation.simulate(Simulation.java:504) at info.openrocket.swing.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:55) at info.openrocket.swing.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:24) at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304) at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343) at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) at java.base/java.lang.Thread.run(Thread.java:840)

---------- Thread information ----------
Thread[AWT-EventQueue-0,6,main]

---------- System information ----------
OpenRocket version: 24.12
OpenRocket source: default
OpenRocket location: C:\Program Files\OpenRocket\.\jar\OpenRocket-24.12.jar
User-defined thrust curves location: C:\Users\DylanCole\AppData\Roaming\OpenRocket\ThrustCurves
LAF: com.formdev.flatlaf.FlatDarculaLaf
JOGL version: null
Current default locale: en_GB
JVM Arguments: -Dexe4j.semaphoreName=Local\c:_program_files_openrocket_openrocket.exe0 -Dexe4j.isInstall4j=true -Dexe4j.moduleName=C:\Program Files\OpenRocket\OpenRocket.exe -Dexe4j.tempDir= -Dexe4j.unextractedPosition=0 -Djava.library.path=c:\program files\openrocket\jre\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\PuTTY\;C:\Program Files\u-center2\utilities;C:\Program Files\Inkscape\bin;C:\Program Files\MATLAB\R2025b\bin;C:\Users\DylanCole\AppData\Local\Microsoft\WindowsApps;C:\Users\DylanCole\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\DylanCole\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\Scripts;C:\Program Files\u-center2\utilities;C:\Users\DylanCole\AppData\Local\GitHubDesktop\bin;C:\Program Files\Tesseract-OCR\; -Dexe4j.consoleCodepage=cp0 -Dapple.awt.application.appearance=system -Dsun.java2d.noddraw=true -Dsun.java2d.d3d=false -Dsun.java2d.ddforcevram=true -Dsun.java2d.ddblit=false -Dswing.useflipBufferStrategy=True --add-exports=java.base/java.lang=ALL-UNNAMED --add-exports=java.desktop/sun.awt=ALL-UNNAMED --add-exports=java.desktop/sun.java2d=ALL-UNNAMED -Dinstall4j.launcherId=59 -Dinstall4j.swt=false -agentpath:C:\Program Files\SentinelOne\Sentinel Agent 26.1.1.163\SentinelJava64.dll=extendedCapabilities
System properties:
apple.awt.application.appearance=system
exe4j.consoleCodepage=cp0
exe4j.isInstall4j=true
exe4j.launchName=C:\Program Files\OpenRocket\OpenRocket.exe
exe4j.moduleName=C:\Program Files\OpenRocket\OpenRocket.exe
exe4j.semaphoreName=Local\c:_program_files_openrocket_openrocket.exe0
exe4j.tempDir=
exe4j.unextractedPosition=0
file.encoding=Cp1252
file.separator=\
flatlaf.uiScale=1.0
install4j.appDir=C:\Program Files\OpenRocket\
install4j.exeDir=C:\Program Files\OpenRocket\
install4j.launcherId=59
install4j.swt=false
java.class.path=C:\Program Files\OpenRocket\.install4j\i4jruntime.jar;C:\Program Files\OpenRocket\.\jar\OpenRocket-24.12.jar
java.class.version=61.0
java.home=c:\program files\openrocket\jre
java.io.tmpdir=C:\Users\DYLANC~1\AppData\Local\Temp\
java.library.path=c:\program files\openrocket\jre\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\PuTTY\;C:\Program Files\u-center2\utilities;C:\Program Files\Inkscape\bin;C:\Program Files\MATLAB\R2025b\bin;C:\Users\DylanCole\AppData\Local\Microsoft\WindowsApps;C:\Users\DylanCole\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\DylanCole\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\Scripts;C:\Program Files\u-center2\utilities;C:\Users\DylanCole\AppData\Local\GitHubDesktop\bin;C:\Program Files\Tesseract-OCR\;
java.runtime.name=OpenJDK Runtime Environment
java.runtime.version=17.0.16+12-LTS
java.specification.maintenance.version=1
java.specification.name=Java Platform API Specification
java.specification.vendor=Oracle Corporation
java.specification.version=17
java.util.Arrays.useLegacyMergeSort=true
java.vendor=BellSoft
java.vendor.url=https://bell-sw.com/
java.vendor.url.bug=https://bell-sw.com/support
java.version=17.0.16
java.version.date=2025-07-15
java.vm.compressedOopsMode=Zero based
java.vm.info=mixed mode
java.vm.name=OpenJDK 64-Bit Server VM
java.vm.specification.name=Java Virtual Machine Specification
java.vm.specification.vendor=Oracle Corporation
java.vm.specification.version=17
java.vm.vendor=BellSoft
java.vm.version=17.0.16+12-LTS
jdk.debug=release
jnlp.jogamp.tmp.cache.root=jln9454369319391528643
line.separator=\u000d\u000a
native.encoding=Cp1252
os.arch=amd64
os.name=Windows 11
os.version=10.0
path.separator=;
sun.arch.data.model=64
sun.awt.enableExtraMouseButtons=true
sun.awt.exception.handler=info.openrocket.swing.gui.main.SwingExceptionHandler$AwtHandler
sun.boot.library.path=c:\program files\openrocket\jre\bin
sun.cpu.endian=little
sun.cpu.isalist=amd64
sun.io.unicode.encoding=UnicodeLittle
sun.java.command=C:\Program Files\OpenRocket\OpenRocket.exe
sun.java2d.d3d=false
sun.java2d.ddblit=false
sun.java2d.ddforcevram=true
sun.java2d.noddraw=true
sun.jnu.encoding=Cp1252
sun.management.compiler=HotSpot 64-Bit Tiered Compilers
sun.os.patch.level=
swing.useflipBufferStrategy=True
user.country=GB
user.dir=C:\Program Files\OpenRocket
user.home=C:\Users\DylanCole
user.language=en
user.name=DylanCole
user.script=
user.timezone=Europe/London
user.variant=
---------- Error log ----------
4449085 0.000 USER (-) ===== 1220 USER lines removed =====
4449085 522994.274 USER (DoubleModel.java:110) SpinnerModel setValue called for DoubleModel[TrapezoidFinSet:CantAngle] newValue=0.0 converted=0.0
4473975 523030.577 USER (DoubleModel.java:110) SpinnerModel setValue called for DoubleModel[Bulkhead:Length] newValue=10.0 converted=0.01
4521889 523050.461 USER (DoubleModel.java:110) SpinnerModel setValue called for DoubleModel[Parachute:AxialOffset] newValue=-600.0 converted=-0.6
4538002 523067.280 USER (DoubleModel.java:110) SpinnerModel setValue called for DoubleModel[ShockCord:AxialOffset] newValue=-585.0 converted=-0.585
4554445 523074.070 USER (DoubleModel.java:110) SpinnerModel setValue called for DoubleModel[MassComponent:AxialOffset] newValue=-585.0 converted=-0.585
4570895 523223.437 USER (DoubleModel.java:110) SpinnerModel setValue called for DoubleModel[MassComponent:Length] newValue=50.0 converted=0.05
4586611 523227.131 USER (DoubleModel.java:110) SpinnerModel setValue called for DoubleModel[MassComponent:Density] newValue=12.8 converted=12800.0
4602096 523235.427 USER (DoubleModel.java:110) SpinnerModel setValue called for DoubleModel[MassComponent:AxialOffset] newValue=-585.0 converted=-0.585
4617599 523245.394 USER (DoubleModel.java:110) SpinnerModel setValue called for DoubleModel[MassComponent:AxialOffset] newValue=-535.0 converted=-0.535
4659596 523267.478 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=23 scaledValue=0.1445132620651305 converted=0.13962634015954636
4659600 523267.596 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=0 scaledValue=0.0 converted=0.0
4659604 523268.379 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=6 scaledValue=0.03769911184307752 converted=0.03490658503988659
4659608 523268.500 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=63 scaledValue=0.39584067435231396 converted=0.4014257279586958
4659612 523268.614 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=94 scaledValue=0.5906194188748811 converted=0.5934119456780721
4659616 523268.733 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=125 scaledValue=0.7853981633974483 converted=0.7853981633974483
4659620 523268.850 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=148 scaledValue=0.9299114254625788 converted=0.9250245035569946
4659624 523268.973 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=159 scaledValue=0.9990264638415542 converted=0.9948376736367679
4659628 523269.100 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=162 scaledValue=1.017876019763093 converted=1.0122909661567112
4659632 523269.225 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=170 scaledValue=1.0681415022205298 converted=1.064650843716541
4659636 523269.343 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=179 scaledValue=1.124690169985146 converted=1.117010721276371
4659640 523269.458 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=193 scaledValue=1.2126547642856602 converted=1.2042771838760873
4659644 523269.579 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=216 scaledValue=1.3571680263507906 converted=1.361356816555577
4659648 523269.694 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=239 scaledValue=1.501681288415921 converted=1.5009831567151235
4659652 523269.811 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=244 scaledValue=1.533097214951819 converted=1.53588974175501
4659656 523269.929 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=250 scaledValue=1.5707963267948966 converted=1.5707963267948966
4659660 523270.072 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=256 scaledValue=1.6084954386379742 converted=1.6057029118347832
4659664 523270.213 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=259 scaledValue=1.6273449945595129 converted=1.6231562043547265
4659668 523270.362 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[RocketFigure:Rotation] newValue=261 scaledValue=1.6399113651738721 converted=1.6406094968746698
4659793 523274.525 USER (GUIUtil.java:245) Closing dialog info.openrocket.swing.gui.configdialog.ComponentConfigDialog[dialog304,211,4,962x484,invalid,layout=java.awt.BorderLayout,MODELESS,title=Nose Cone configuration,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,7,1,948x476,invalid,layout=com.formdev.flatlaf.ui.FlatRootPaneUI$FlatRootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true]
4660308 523280.860 USER (GUIUtil.java:245) Closing dialog info.openrocket.swing.gui.configdialog.ComponentConfigDialog[dialog305,211,4,962x484,invalid,layout=java.awt.BorderLayout,MODELESS,title=Nose Cone configuration,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,7,1,948x476,invalid,layout=com.formdev.flatlaf.ui.FlatRootPaneUI$FlatRootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true]
4678342 523298.029 USER (DoubleModel.java:110) SpinnerModel setValue called for DoubleModel[NoseCone:Length] newValue=1.0 converted=0.001
4678350 523300.031 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[NoseCone:ShapeParameter] newValue=513 scaledValue=0.513 converted=0.5
4678352 523300.037 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[NoseCone:ShapeParameter] newValue=538 scaledValue=0.538 converted=0.5
4678354 523300.045 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[NoseCone:ShapeParameter] newValue=563 scaledValue=0.563 converted=0.6
4678362 523300.224 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[NoseCone:ShapeParameter] newValue=1000 scaledValue=1.0 converted=1.0
4678370 523300.945 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[NoseCone:ShapeParameter] newValue=975 scaledValue=0.975 converted=1.0
4678372 523300.951 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[NoseCone:ShapeParameter] newValue=913 scaledValue=0.913 converted=0.9
4678380 523301.140 USER (DoubleModel.java:373) SliderModel setValue called for DoubleModel[NoseCone:ShapeParameter] newValue=0 scaledValue=0.0 converted=0.0
4679530 523603.308 USER (PresetModel.java:67) User selected preset item 'CF-24' for component Parachute, 18 in., nylon, 6 lines
4679550 0.000 INFO (-) ===== 3378 INFO lines removed =====
4679550 523603.646 INFO (SimulationWorker.java:55) >>Starting simulation of branch: Payload
4679554 523603.647 INFO (SimulationWorker.java:55) Queueing Ignition Event for: Inner Tube/c8bcfd2e / 2946L820-P - Armed @: 0.0
4679560 523603.647 INFO (SimulationWorker.java:55) Igniting motor: Inner Tube/c8bcfd2e / 2946L820-P - Armed @0.0
4679690 523606.636 USER (BooleanModel.java:303) Value of BooleanModel[Parachute:MassOverridden] changed to true oldValue=false
4679692 523606.660 INFO (RocketPanel.java:1016) using pre-existing simulation
4679700 523606.976 INFO (SimulationWorker.java:55) >>Starting simulation of branch: Payload
4679704 523606.978 INFO (SimulationWorker.java:55) Queueing Ignition Event for: Inner Tube/c8bcfd2e / 2946L820-P - Armed @: 0.0
4679710 523606.979 INFO (SimulationWorker.java:55) Igniting motor: Inner Tube/c8bcfd2e / 2946L820-P - Armed @0.0
4679842 523609.126 USER (DoubleModel.java:110) SpinnerModel setValue called for DoubleModel[Parachute:OverrideMass] newValue=104.0 converted=0.10400000000000001
4679844 523609.154 INFO (RocketPanel.java:1016) using pre-existing simulation
4679852 523609.469 INFO (SimulationWorker.java:55) >>Starting simulation of branch: Payload
4679856 523609.470 INFO (SimulationWorker.java:55) Queueing Ignition Event for: Inner Tube/c8bcfd2e / 2946L820-P - Armed @: 0.0
4679862 523609.470 INFO (SimulationWorker.java:55) Igniting motor: Inner Tube/c8bcfd2e / 2946L820-P - Armed @0.0
4680446 523612.339 INFO (ComponentConfigDialog.java:278) Adding undo position 'Modify Parachute' to info.openrocket.core.document.OpenRocketDocument@14fc672c, document is in unclean state
4680447 523639.005 USER (GUIUtil.java:245) Closing dialog info.openrocket.swing.gui.configdialog.ComponentConfigDialog[dialog310,211,4,962x605,invalid,layout=java.awt.BorderLayout,MODELESS,title=Parachute configuration,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,7,1,948x597,invalid,layout=com.formdev.flatlaf.ui.FlatRootPaneUI$FlatRootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true]
4680899 523641.090 INFO (ComponentConfigDialog.java:278) Adding undo position 'Modify Parachute' to info.openrocket.core.document.OpenRocketDocument@14fc672c, document was in clean state
4680900 523660.413 USER (PresetModel.java:67) User selected preset item 'PAR-18TM' for component Parachute, 18 in., nylon, 6 lines
4680901 523660.413 INFO (PresetModel.java:74) Adding undo position 'Use Preset Parachute' to info.openrocket.core.document.OpenRocketDocument@14fc672c, document was in clean state
4680905 523660.443 INFO (RocketPanel.java:1016) using pre-existing simulation
4680962 523660.768 INFO (SimulationWorker.java:55) >>Starting simulation of branch: Payload
4680966 523660.773 INFO (SimulationWorker.java:55) Queueing Ignition Event for: Inner Tube/c8bcfd2e / 2946L820-P - Armed @: 0.0
4680972 523660.773 INFO (SimulationWorker.java:55) Igniting motor: Inner Tube/c8bcfd2e / 2946L820-P - Armed @0.0
4681105 523664.266 USER (BooleanModel.java:303) Value of BooleanModel[Parachute:MassOverridden] changed to true oldValue=false
4681107 523664.297 INFO (RocketPanel.java:1016) using pre-existing simulation
4681115 523664.610 INFO (SimulationWorker.java:55) >>Starting simulation of branch: Payload
4681119 523664.612 INFO (SimulationWorker.java:55) Queueing Ignition Event for: Inner Tube/c8bcfd2e / 2946L820-P - Armed @: 0.0
4681125 523664.612 INFO (SimulationWorker.java:55) Igniting motor: Inner Tube/c8bcfd2e / 2946L820-P - Armed @0.0
4681260 523668.364 USER (DoubleModel.java:110) SpinnerModel setValue called for DoubleModel[Parachute:OverrideMass] newValue=307.0 converted=0.307
4681262 523668.391 INFO (RocketPanel.java:1016) using pre-existing simulation
4681271 523668.706 INFO (SimulationWorker.java:55) >>Starting simulation of branch: Payload
4681275 523668.707 INFO (SimulationWorker.java:55) Queueing Ignition Event for: Inner Tube/c8bcfd2e / 2946L820-P - Armed @: 0.0
4681281 523668.707 INFO (SimulationWorker.java:55) Igniting motor: Inner Tube/c8bcfd2e / 2946L820-P - Armed @0.0
4681857 523681.236 INFO (ComponentConfigDialog.java:278) Adding undo position 'Modify Mass Component' to info.openrocket.core.document.OpenRocketDocument@14fc672c, document is in unclean state
4681859 523702.095 USER (DoubleModel.java:110) SpinnerModel setValue called for DoubleModel[MassComponent:Length] newValue=200.0 converted=0.2
4681862 523702.129 INFO (RocketPanel.java:1016) using pre-existing simulation
4681871 523702.445 INFO (SimulationWorker.java:55) >>Starting simulation of branch: Payload
4681875 523702.447 INFO (SimulationWorker.java:55) Queueing Ignition Event for: Inner Tube/c8bcfd2e / 2946L820-P - Armed @: 0.0
4681881 523702.447 INFO (SimulationWorker.java:55) Igniting motor: Inner Tube/c8bcfd2e / 2946L820-P - Armed @0.0
4682018 523706.476 USER (DoubleModel.java:110) SpinnerModel setValue called for DoubleModel[MassComponent:Density] newValue=12.8 converted=12800.0
4682020 523706.510 INFO (RocketPanel.java:1016) using pre-existing simulation
4682030 523706.821 INFO (SimulationWorker.java:55) >>Starting simulation of branch: Payload
4682034 523706.822 INFO (SimulationWorker.java:55) Queueing Ignition Event for: Inner Tube/c8bcfd2e / 2946L820-P - Armed @: 0.0
4682040 523706.823 INFO (SimulationWorker.java:55) Igniting motor: Inner Tube/c8bcfd2e / 2946L820-P - Armed @0.0
4682178 523730.088 USER (DoubleModel.java:110) SpinnerModel setValue called for DoubleModel[MassComponent:AxialOffset] newValue=-385.0 converted=-0.385
4682180 523730.123 INFO (RocketPanel.java:1016) using pre-existing simulation
4682189 523730.431 INFO (SimulationWorker.java:55) >>Starting simulation of branch: Payload
4682193 523730.434 INFO (SimulationWorker.java:55) Queueing Ignition Event for: Inner Tube/c8bcfd2e / 2946L820-P - Armed @: 0.0
4682199 523730.434 INFO (SimulationWorker.java:55) Igniting motor: Inner Tube/c8bcfd2e / 2946L820-P - Armed @0.0
4682346 0.000 WARN (-) ===== 21857 WARN lines removed =====
4682346 523731.915 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@230f70eb but it was not registered
4682351 523731.915 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@a631ff3 while still having listeners [javax.swing.JSlider$ModelListener@731161fb, javax.swing.plaf.basic.BasicSliderUI$Handler@4b051833]
4682355 523731.915 WARN (ModelInvalidator.java:82) info.openrocket.swing.gui.adaptors.ModelInvalidator@a631ff3 removing listener that was not registered: javax.swing.JSlider$ModelListener@731161fb
4682357 523731.915 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@72a87557 but it was not registered
4682362 523731.915 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@1521ef37 while still having listeners [javax.swing.JSlider$ModelListener@38f8d770, javax.swing.plaf.basic.BasicSliderUI$Handler@22badf0f]
4682366 523731.915 WARN (ModelInvalidator.java:82) info.openrocket.swing.gui.adaptors.ModelInvalidator@1521ef37 removing listener that was not registered: javax.swing.JSlider$ModelListener@38f8d770
4682368 523731.915 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@3bd5078a but it was not registered
4682372 523731.916 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@68b4c6fc but it was not registered
4682387 523731.916 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@23b0d09c while still having listeners [javax.swing.JSlider$ModelListener@4574037b, javax.swing.plaf.basic.BasicSliderUI$Handler@41ca92a1]
4682391 523731.916 WARN (ModelInvalidator.java:82) info.openrocket.swing.gui.adaptors.ModelInvalidator@23b0d09c removing listener that was not registered: javax.swing.JSlider$ModelListener@4574037b
4682408 523731.916 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@4d8607ec while still having listeners [javax.swing.JSlider$ModelListener@5f6e7d6e, javax.swing.plaf.basic.BasicSliderUI$Handler@1a8c2792]
4682412 523731.916 WARN (ModelInvalidator.java:82) info.openrocket.swing.gui.adaptors.ModelInvalidator@4d8607ec removing listener that was not registered: javax.swing.JSlider$ModelListener@5f6e7d6e
4682419 523731.916 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@23b36e97 while still having listeners [javax.swing.JSlider$ModelListener@32a2ac5e, javax.swing.plaf.basic.BasicSliderUI$Handler@39830b4b]
4682423 523731.916 WARN (ModelInvalidator.java:82) info.openrocket.swing.gui.adaptors.ModelInvalidator@23b36e97 removing listener that was not registered: javax.swing.JSlider$ModelListener@32a2ac5e
4682433 523731.916 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@520be3fa but it was not registered
4682438 523731.916 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@661c0c48 while still having listeners [javax.swing.JSlider$ModelListener@7c9d0cdd, javax.swing.plaf.basic.BasicSliderUI$Handler@7b850de2]
4682442 523731.916 WARN (ModelInvalidator.java:82) info.openrocket.swing.gui.adaptors.ModelInvalidator@661c0c48 removing listener that was not registered: javax.swing.JSlider$ModelListener@7c9d0cdd
4682444 523731.916 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@41fe6976 but it was not registered
4682449 523731.916 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@4e4ab532 while still having listeners [javax.swing.JSlider$ModelListener@3b291829, javax.swing.plaf.basic.BasicSliderUI$Handler@40c84d3]
4682453 523731.916 WARN (ModelInvalidator.java:82) info.openrocket.swing.gui.adaptors.ModelInvalidator@4e4ab532 removing listener that was not registered: javax.swing.JSlider$ModelListener@3b291829
4682455 523731.916 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@1d734163 but it was not registered
4682460 523731.916 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@55076315 while still having listeners [javax.swing.JSlider$ModelListener@6b7e0ff0, javax.swing.plaf.basic.BasicSliderUI$Handler@7145fff5]
4682464 523731.916 WARN (ModelInvalidator.java:82) info.openrocket.swing.gui.adaptors.ModelInvalidator@55076315 removing listener that was not registered: javax.swing.JSlider$ModelListener@6b7e0ff0
4682466 523731.916 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@35cd4363 but it was not registered
4682471 523731.916 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@2fb6886b while still having listeners [javax.swing.JSlider$ModelListener@47223d93, javax.swing.plaf.basic.BasicSliderUI$Handler@4e271140, info.openrocket.swing.gui.adaptors.DoubleModel$AutomaticActionModel@3a8c0845, info.openrocket.swing.gui.adaptors.DoubleModel$AutomaticActionModel@3a8c0845]
4682475 523731.916 WARN (ModelInvalidator.java:82) info.openrocket.swing.gui.adaptors.ModelInvalidator@2fb6886b removing listener that was not registered: javax.swing.JSlider$ModelListener@47223d93
4682477 523731.916 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@71526be3 but it was not registered
4682480 523731.916 WARN (ModelInvalidator.java:82) info.openrocket.swing.gui.adaptors.ModelInvalidator@2fb6886b removing listener that was not registered: info.openrocket.swing.gui.adaptors.DoubleModel$AutomaticActionModel@3a8c0845
4682482 523731.916 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@1e18022d but it was not registered
4682487 523731.916 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@1805dbe1 while still having listeners [javax.swing.JSlider$ModelListener@7ffda752, javax.swing.plaf.basic.BasicSliderUI$Handler@646a1432]
4682491 523731.916 WARN (ModelInvalidator.java:82) info.openrocket.swing.gui.adaptors.ModelInvalidator@1805dbe1 removing listener that was not registered: javax.swing.JSlider$ModelListener@7ffda752
4682493 523731.916 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@79448e13 but it was not registered
4682498 523731.916 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@2f8eec46 while still having listeners [javax.swing.JSlider$ModelListener@63e0d6b7, javax.swing.plaf.basic.BasicSliderUI$Handler@3abeeb6e]
4682502 523731.916 WARN (ModelInvalidator.java:82) info.openrocket.swing.gui.adaptors.ModelInvalidator@2f8eec46 removing listener that was not registered: javax.swing.JSlider$ModelListener@63e0d6b7
4682504 523731.916 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@58955179 but it was not registered
4682509 523731.916 WARN (ModelInvalidator.java:105) Invalidating info.openrocket.swing.gui.adaptors.ModelInvalidator@57a861d0 while still having listeners [javax.swing.JSlider$ModelListener@71f8022a, javax.swing.plaf.basic.BasicSliderUI$Handler@ce6740e]
4682513 523731.916 WARN (ModelInvalidator.java:82) info.openrocket.swing.gui.adaptors.ModelInvalidator@57a861d0 removing listener that was not registered: javax.swing.JSlider$ModelListener@71f8022a
4682515 523731.916 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@1a000c65 but it was not registered
4682528 523731.916 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@230f70eb but it was not registered
4682540 523731.916 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@72a87557 but it was not registered
4682552 523731.916 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@3bd5078a but it was not registered
4682575 0.000 DEBUG (-) ===== 48523 DEBUG lines removed =====
4682575 523731.917 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[AppearanceBuilder:OffsetU]
4682576 523731.917 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@5b229d4e
4682579 523731.917 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[AppearanceBuilder:OffsetV]
4682580 523731.917 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@1936e948
4682583 523731.917 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[AppearanceBuilder:Opacity]
4682584 523731.917 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@4d8607ec
4682587 523731.917 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[AppearanceBuilder:Rotation]
4682588 523731.917 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@23b36e97
4682591 523731.917 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: EnumModel[info.openrocket.core.appearance.AppearanceBuilder:EdgeMode]
4682592 523731.917 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@3a5d83d5
4682594 523731.917 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@520be3fa but it was not registered
4682595 523731.917 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: EnumModel[info.openrocket.core.rocketcomponent.MassComponent:MassComponentType]
4682596 523731.917 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@41760686
4682598 523731.917 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@41fe6976 but it was not registered
4682599 523731.917 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[MassComponent:ComponentMass]
4682600 523731.917 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@661c0c48
4682602 523731.917 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@1d734163 but it was not registered
4682603 523731.917 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[MassComponent:Density]
4682604 523731.917 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@4e4ab532
4682606 523731.917 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@35cd4363 but it was not registered
4682607 523731.917 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[MassComponent:Length]
4682608 523731.917 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@55076315
4682610 523731.917 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@71526be3 but it was not registered
4682611 523731.917 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[MassComponent:Radius]
4682612 523731.917 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@2fb6886b
4682614 523731.917 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@1e18022d but it was not registered
4682615 523731.917 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: EnumModel[info.openrocket.core.rocketcomponent.MassComponent:AxialMethod]
4682616 523731.917 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@49e9145
4682618 523731.917 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@79448e13 but it was not registered
4682619 523731.917 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[MassComponent:AxialOffset]
4682620 523731.917 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@1805dbe1
4682622 523731.917 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@58955179 but it was not registered
4682623 523731.917 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[MassComponent:RadialPosition]
4682624 523731.917 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@2f8eec46
4682626 523731.917 WARN (ModelInvalidator.java:109) Attempted to remove ComponentChangeListener info.openrocket.core.util.ComponentChangeAdapter@1a000c65 but it was not registered
4682627 523731.917 DEBUG (ModelInvalidator.java:113) Adding object into collectable list: DoubleModel[MassComponent:RadialDirection]
4682628 523731.917 DEBUG (ModelInvalidator.java:114) Adding object into collectable list: info.openrocket.swing.gui.adaptors.ModelInvalidator@57a861d0
4682634 523739.495 DEBUG (SimulationWorker.java:55) Simulation: calling simulator
4682636 523739.499 INFO (SimulationWorker.java:55) >>Starting simulation of branch: Payload
4682640 523739.501 INFO (SimulationWorker.java:55) Queueing Ignition Event for: Inner Tube/c8bcfd2e / 5015L1115-P - Armed @: 0.0
4682642 523739.501 DEBUG (SimulationRunDialog.java:479) LAUNCH
4682647 523739.502 INFO (SimulationWorker.java:55) Igniting motor: Inner Tube/c8bcfd2e / 5015L1115-P - Armed @0.0
4682653 523739.502 DEBUG (SimulationWorker.java:55) Computed flight values: maxAltitude=0.0 maxVelocity=0.0 maxAcceleration=NaN maxMachNumber=NaN timeToApogee=0.0 flightTime=0.0 groundHitVelocity=NaN launchRodVelocity=NaN optimumDelay=NaN
4682654 523739.538 DEBUG (SimulationRunDialog.java:240) Progressbar value 0
4682655 523739.538 DEBUG (SimulationRunDialog.java:245) No simulation status data available, setting empty labels
4682656 523739.575 DEBUG (SimulationRunDialog.java:364) simulationStage boost: t=0.0 est=4.466713356316449
4682657 523739.575 DEBUG (SimulationRunDialog.java:456) Setting progress to 0 (real 0)
4682658 523739.575 DEBUG (SimulationRunDialog.java:240) Progressbar value 0
4682659 523739.576 ERROR (SwingExceptionHandler.java:120) An exception occurred during the simulation
> info.openrocket.core.util.BugException: BUG: Simulation resulted in not-a-number (NaN) value for structureMass.getCenterOfMass(), please report a bug, c=(NaN,NaN,NaN,w=6.50156) > at info.openrocket.core.simulation.AbstractSimulationStepper.checkNaN(AbstractSimulationStepper.java:266) > at info.openrocket.core.simulation.AbstractSimulationStepper.calculateStructureMass(AbstractSimulationStepper.java:217) > at info.openrocket.core.simulation.RK4SimulationStepper.computeAcceleration(RK4SimulationStepper.java:337) > at info.openrocket.core.simulation.RK4SimulationStepper.calculateAcceleration(RK4SimulationStepper.java:284) > at info.openrocket.core.simulation.RK4SimulationStepper.computeParameters(RK4SimulationStepper.java:261) > at info.openrocket.core.simulation.RK4SimulationStepper.step(RK4SimulationStepper.java:113) > at info.openrocket.core.simulation.BasicEventSimulationEngine.simulateLoop(BasicEventSimulationEngine.java:198) > at info.openrocket.core.simulation.BasicEventSimulationEngine.simulate(BasicEventSimulationEngine.java:121) > at info.openrocket.core.document.Simulation.simulate(Simulation.java:504) > at info.openrocket.swing.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:55) > at info.openrocket.swing.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:24) > at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304) > at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) > at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343) > at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) > at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) > at java.base/java.lang.Thread.run(Thread.java:840)
4682660 523739.576 INFO (SwingExceptionHandler.java:143) Running in EDT, showing dialog
4682661 523739.576 INFO (SwingExceptionHandler.java:202) Showing Exception dialog
4682662 523742.120 USER (SwingExceptionHandler.java:216) User chose not to fill bug report
4682663 523742.120 DEBUG (SimulationRunDialog.java:407) Simulation done
4682664 523742.120 DEBUG (SimulationRunDialog.java:456) Setting progress to 100 (real 100)
4682665 523742.120 DEBUG (SimulationRunDialog.java:228) Everything done.
4682666 523742.168 INFO (SimulationPanel.java:501) Running simulations took 2676 ms
4682667 523742.264 DEBUG (GUIUtil.java:312) Clearing all models of window info.openrocket.swing.gui.simulation.SimulationRunDialog[dialog314,588,340,360x231,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=Running simulations…,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,7,1,346x223,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true]
4682668 523742.264 DEBUG (GUIUtil.java:314) Adding object into collectable list: info.openrocket.swing.gui.simulation.SimulationRunDialog[dialog314,588,340,360x231,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=Running simulations…,defaultCloseOperation=DISPOSE_ON_CLOSE,rootPane=javax.swing.JRootPane[,7,1,346x223,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=16777673,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true]
4682672 523750.988 DEBUG (SimulationWorker.java:55) Simulation: calling simulator
4682674 523750.991 INFO (SimulationWorker.java:55) >>Starting simulation of branch: Payload
4682678 523750.992 INFO (SimulationWorker.java:55) Queueing Ignition Event for: Inner Tube/c8bcfd2e / 2946L820-P - Armed @: 0.0
4682680 523750.992 DEBUG (SimulationRunDialog.java:479) LAUNCH
4682685 523750.993 INFO (SimulationWorker.java:55) Igniting motor: Inner Tube/c8bcfd2e / 2946L820-P - Armed @0.0
4682691 523750.994 DEBUG (SimulationWorker.java:55) Computed flight values: maxAltitude=0.0 maxVelocity=0.0 maxAcceleration=NaN maxMachNumber=NaN timeToApogee=0.0 flightTime=0.0 groundHitVelocity=NaN launchRodVelocity=NaN optimumDelay=NaN
4682773 0.000 VBOSE (-) ===== 4607602 VBOSE lines removed =====
4682773 523751.004 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682774 523751.004 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682775 523751.004 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682776 523751.004 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682777 523751.004 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682778 523751.005 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682779 523751.005 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682780 523751.005 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682781 523751.005 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682782 523751.005 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682783 523751.005 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682784 523751.005 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682785 523751.005 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682786 523751.005 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682787 523751.005 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682788 523751.005 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682789 523751.005 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682790 523751.005 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682791 523751.005 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682792 523751.005 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682793 523751.005 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682794 523751.006 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682795 523751.006 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682796 523751.006 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682797 523751.006 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682798 523751.006 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682799 523751.006 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682800 523751.006 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682801 523751.006 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682802 523751.006 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682803 523751.006 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682804 523751.006 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682805 523751.006 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682806 523751.006 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682807 523751.006 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682808 523751.007 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682809 523751.007 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682810 523751.007 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682811 523751.007 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682812 523751.007 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682813 523751.007 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682814 523751.007 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682815 523751.007 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682816 523751.007 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682817 523751.007 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682818 523751.007 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682819 523751.008 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682820 523751.008 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682821 523751.008 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682822 523751.008 VBOSE (AbstractScaleFigure.java:209) ____ Updating RocketFigure to: 0.139227 user scale, 657.767 overall scale
4682823 523751.011 DEBUG (SimulationRunDialog.java:240) Progressbar value 0
4682824 523751.011 DEBUG (SimulationRunDialog.java:245) No simulation status data available, setting empty labels
4682825 523751.124 DEBUG (SimulationRunDialog.java:364) simulationStage boost: t=0.0 est=3.5882216092355885
4682826 523751.124 DEBUG (SimulationRunDialog.java:456) Setting progress to 0 (real 0)
4682827 523751.124 DEBUG (SimulationRunDialog.java:240) Progressbar value 0
4682828 523751.124 ERROR (SwingExceptionHandler.java:120) An exception occurred during the simulation
> info.openrocket.core.util.BugException: BUG: Simulation resulted in not-a-number (NaN) value for structureMass.getCenterOfMass(), please report a bug, c=(NaN,NaN,NaN,w=6.50156) > at info.openrocket.core.simulation.AbstractSimulationStepper.checkNaN(AbstractSimulationStepper.java:266) > at info.openrocket.core.simulation.AbstractSimulationStepper.calculateStructureMass(AbstractSimulationStepper.java:217) > at info.openrocket.core.simulation.RK4SimulationStepper.computeAcceleration(RK4SimulationStepper.java:337) > at info.openrocket.core.simulation.RK4SimulationStepper.calculateAcceleration(RK4SimulationStepper.java:284) > at info.openrocket.core.simulation.RK4SimulationStepper.computeParameters(RK4SimulationStepper.java:261) > at info.openrocket.core.simulation.RK4SimulationStepper.step(RK4SimulationStepper.java:113) > at info.openrocket.core.simulation.BasicEventSimulationEngine.simulateLoop(BasicEventSimulationEngine.java:198) > at info.openrocket.core.simulation.BasicEventSimulationEngine.simulate(BasicEventSimulationEngine.java:121) > at info.openrocket.core.document.Simulation.simulate(Simulation.java:504) > at info.openrocket.swing.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:55) > at info.openrocket.swing.gui.simulation.SimulationWorker.doInBackground(SimulationWorker.java:24) > at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304) > at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) > at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343) > at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) > at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) > at java.base/java.lang.Thread.run(Thread.java:840)
4682829 523751.124 INFO (SwingExceptionHandler.java:143) Running in EDT, showing dialog
4682830 523751.124 INFO (SwingExceptionHandler.java:202) Showing Exception dialog
4682831 523754.148 USER (SwingExceptionHandler.java:221) User requested sending bug report
4682832 523754.163 DEBUG (BugReportDialog.java:230) Found jar file using codeSource
---------- End of bug report ----------

--- a/core/src/main/java/info/openrocket/core/rocketcomponent/SymmetricComponent.java
+++ b/core/src/main/java/info/openrocket/core/rocketcomponent/SymmetricComponent.java
@@ -492,7 +492,10 @@ public abstract class SymmetricComponent extends BodyComponent implements BoxBou
 
 			final double dFullV = fullCG.getWeight();
 			final double dV = fullCG.getWeight() - innerCG.getWeight();
-			final double dCG = (fullCG.getX() * fullCG.getWeight() - innerCG.getX() * innerCG.getWeight()) / dV;
+			// Some discontinuous profiles contain slices with no material.  Their CG is
+			// irrelevant, but it must remain finite so that 0 * dCG does not become NaN.
+			final double dCG = dV == 0.0 ? l / 2.0
+					: (fullCG.getX() * fullCG.getWeight() - innerCG.getX() * innerCG.getWeight()) / dV;
 
 			// First moment, used later for CG calculation
 			final double dCGx = dV * (x1 + dCG);

--- a/core/src/test/java/info/openrocket/core/simulation/BansheeMk2SimulationTest.java
+++ b/core/src/test/java/info/openrocket/core/simulation/BansheeMk2SimulationTest.java
@@ -1,0 +1,58 @@
+package info.openrocket.core.simulation;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import org.junit.jupiter.api.Test;
+
+import info.openrocket.core.document.Simulation;
+import info.openrocket.core.rocketcomponent.BodyTube;
+import info.openrocket.core.rocketcomponent.NoseCone;
+import info.openrocket.core.rocketcomponent.PodSet;
+import info.openrocket.core.rocketcomponent.Rocket;
+import info.openrocket.core.rocketcomponent.Transition;
+import info.openrocket.core.util.BaseTestCase;
+import info.openrocket.core.util.TestRockets;
+
+/**
+ * Regression coverage for GitHub issue #3161, reported with the Banshee Mk2
+ * design by GitHub user Enderdyls (Dylan Cole).
+ */
+public class BansheeMk2SimulationTest extends BaseTestCase {
+
+	/**
+	 * Verifies that zero-volume integration slices in a short Power-series tail
+	 * cone do not contaminate the complete rocket's center of mass with NaN.
+	 */
+	@Test
+	public void testSimulationWithZeroVolumeTailConeSlices() {
+		Rocket rocket = TestRockets.makeEstesAlphaIII();
+		BodyTube bodyTube = (BodyTube) rocket.getStage(0).getChild(1);
+
+		PodSet podSet = new PodSet();
+		podSet.setInstanceCount(1);
+		bodyTube.addChild(podSet);
+
+		NoseCone podNoseCone = new NoseCone(Transition.Shape.HAACK, 0.15, 0.011);
+		podNoseCone.setShapeParameter(0.0);
+		podSet.addChild(podNoseCone);
+
+		// These are the dimensions and shape settings that triggered the original failure.
+		NoseCone tailCone = new NoseCone(Transition.Shape.POWER, 0.001, 0.011);
+		tailCone.setShapeParameter(0.0);
+		tailCone.setThickness(0.002);
+		tailCone.setFlipped(true);
+		podSet.addChild(tailCone);
+
+		Simulation simulation = new Simulation(rocket);
+		simulation.setFlightConfigurationId(TestRockets.TEST_FCID_0);
+		simulation.getOptions().setISAAtmosphere(true);
+		simulation.getOptions().setTimeStep(0.05);
+		simulation.getOptions().setRandomSeed(0x3161);
+
+		assertDoesNotThrow(() -> {
+			simulation.simulate();
+		});
+		assertFalse(simulation.hasErrors(), simulation::getStatusDescription);
+	}
+}

